### PR TITLE
add queryable extension to ERC721Base

### PR DESCRIPTION
**Pull request by WhiteOakKong**

### CODE REVIEW

File: contracts/eip/queryable/ERC721AQueryable.sol [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+pragma solidity ^0.8.4;
+
+import "./IERC721AQueryable.sol";
+import "../ERC721AVirtualApprove.sol";
+
+/**
+ * @title ERC721A Queryable
+ * @dev ERC721A subclass with convenience query functions.
+ */
+abstract contract ERC721AQueryable is ERC721A, IERC721AQueryable {
```
Comment (joaquim-verges) [inline contracts/eip/queryable/ERC721AQueryable.sol:14]: don't we already have this file somewhere?

Comment (WhiteOakKong) [reply contracts/eip/queryable/ERC721AQueryable.sol:14]: > don't we already have this file somewhere?

There isn't a non-upgradable version currently.

Comment (joaquim-verges) [reply contracts/eip/queryable/ERC721AQueryable.sol:14]: Gotcha

**Review by joaquim-verges — APPROVED** [review]
How's the contract sizes? Bases should have a bit of wiggle room to add custom fns

**Comment by WhiteOakKong** [comment]
> How's the contract sizes? Bases should have a bit of wiggle room to add custom fns

ERC721Base has 13kb in left over space. The queryable adds ~1kb.

**Comment by joaquim-verges** [comment]
Should be plenty. Ship it

**Comment by joaquim-verges** [comment]
@nkrishang @kumaryash90 can you review?

**Pull request merged**